### DISCUSSION TITLE
A StateNotifier per listview row

### DISCUSSION THREAD

**Discussion by Sadmansamee** [rrousselGit/riverpod#2131] [Q&A]
I have a listview where each row has some internal business logic and micro-interactions going on, just like it this item is favorite or not, on a toggle button "isFavorite" state changes, and does an async task (saves of Prefs), now how do I do it or have a dedicated StateNotifier for this logic? In my case each row of listview would have a StateNotifier provider.

**Comment by rrousselGit** [comment]
You'd typically use "family" to have a different state per row

**Reply by Sadmansamee** [reply]
got it
